tests/nested/core/core20-factory-reset/task.yaml

```diff
@@ -0,0 +1,87 @@
+summary: Verify factory reset of a UC20 system
+
+details: |
+    This test checks that UC20 can be reset to factory state
+
+systems: [ubuntu-20.04-64]
+
+environment:
+    NESTED_ENABLE_SECURE_BOOT: false
+    NESTED_ENABLE_TPM: false
+
+execute: |
+    echo "Wait for the system to be seeded first"
+    tests.nested exec "sudo snap wait system seed.loaded"
+
+    tests.nested exec snap model > initial-model
+    tests.nested exec snap model --serial > initial-serial
+    tests.nested exec sudo blkid |grep -v /dev/loop > initial-disk
+
+    echo "Request factory reset"
+    boot_id=$(tests.nested boot-id)
+
+    # leave some marker files
+    tests.nested exec sudo touch /run/mnt/ubuntu-seed/marker
+    tests.nested exec sudo touch /run/mnt/ubuntu-save/marker
+    tests.nested exec sudo touch /run/mnt/ubuntu-boot/marker
+    tests.nested exec sudo touch /writable/marker
+
+    # add || true in case the SSH connection is broken while executing this
+    # since this command causes an immediate reboot
+    tests.nested exec "sudo snap reboot --factory-reset" || true
+
+    tests.nested wait-for reboot "${boot_id}"
+
+    # check that we are back in run mode
+    tests.nested exec cat /proc/cmdline | MATCH 'snapd_recovery_mode=run'
+
+    # wait for the system to get setup and finish seeding
+    tests.nested wait-for snap-command
+    tests.nested exec "sudo snap wait system seed.loaded"
+
+    # wait up to two minutes for serial registration
+    retry -n 60 --wait 2 tests.nested exec snap model --serial
+
+    # post factory reset world
+
+    tests.nested exec snap model > current-model
+    tests.nested exec snap model --serial > current-serial
+    tests.nested exec sudo blkid |grep -v /dev/loop > current-disk
+    # serials should be identical
+    diff -up initial-model current-model
+    diff -up initial-serial current-serial
```

Comment:
The `-p` option seems out of place, unless I miss something.

Reply:
I removed the -p now